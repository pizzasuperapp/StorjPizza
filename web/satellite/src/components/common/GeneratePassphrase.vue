// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="encrypt-container">
        <EncryptIcon />
        <h1 class="encrypt-container__title" aria-roledescription="enc-title">Encryption passphrase</h1>
        <p class="encrypt-container__info">
            The encryption passphrase is used to encrypt and access the data that you upload to Storj.
        </p>
        <div class="encrypt-container__functional">
            <div class="encrypt-container__functional__header">
                <p class="encrypt-container__functional__header__gen" :class="{ active: isGenerate }" @click="setToGenerate">
                    Generate a new passphrase
                </p>
                <div class="encrypt-container__functional__header__right" :class="{ active: !isGenerate }">
                    <p
                        class="encrypt-container__functional__header__right__enter"
                        :class="{ active: !isGenerate }"
                        aria-roledescription="enter-passphrase-label"
                        @click="setToEnter"
                    >
                        Enter your own passphrase
                    </p>
                    <VInfo class="encrypt-container__functional__header__right__info-button">
                        <template #icon>
                            <InfoIcon class="encrypt-container__functional__header__right__info-button__image" :class="{ active: !isGenerate }" />
                        </template>
                        <template #message>
                            <p class="encrypt-container__functional__header__right__info-button__message">
                                We strongly encourage you to use a mnemonic phrase, which is automatically generated on
                                the client-side for you. Alternatively, you can enter your own passphrase.
                            </p>
                        </template>
                    </VInfo>
                </div>
            </div>
            <div v-if="isGenerate" class="encrypt-container__functional__generate">
                <p class="encrypt-container__functional__generate__value">{{ passphrase }}</p>
                <VButton
                    class="encrypt-container__functional__generate__button"
                    label="Copy"
                    width="66px"
                    height="30px"
                    is-blue-white="true"
                    is-uppercase="true"
                    :on-press="onCopyClick"
                />
            </div>
            <div v-else class="encrypt-container__functional__enter">
                <VInput
                    label="Your Passphrase"
                    placeholder="Enter a passphrase here..."
                    :error="enterError"
                    role-description="passphrase"
                    is-password="true"
                    :disabled="isLoading"
                    @setData="setPassphrase"
                />
            </div>
            <h2 class="encrypt-container__functional__warning-title" aria-roledescription="warning-title">
                Save your encryption passphrase
            </h2>
            <p class="encrypt-container__functional__warning-msg">
                Please note that Storj does not know or store your encryption passphrase. If you lose it, you will not
                be able to recover your files.
            </p>
            <p class="encrypt-container__functional__download" @click="onDownloadClick">Download as a text file</p>
            <VCheckbox
                class="encrypt-container__functional__checkbox"
                label="I understand, and I have saved the passphrase."
                :is-checkbox-error="isCheckboxError"
                @setData="setSavingConfirmation"
            />
        </div>
        <div class="encrypt-container__buttons">
            <VButton
                v-if="isNewObjectsFlow"
                class="encrypt-container__buttons__back"
                label="< Back"
                height="64px"
                border-radius="62px"
                is-blue-white="true"
                :on-press="onBackClick"
                :is-disabled="isLoading"
            />
            <VButton
                label="Next >"
                height="64px"
                border-radius="62px"
                :on-press="onNextButtonClick"
                :is-disabled="isLoading || !isSavingConfirmed"
            />
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { generateMnemonic } from "bip39";

import { LocalData, UserIDPassSalt } from "@/utils/localData";
import { Download } from "@/utils/download";

import VButton from '@/components/common/VButton.vue';
import VInfo from "@/components/common/VInfo.vue";
import VInput from "@/components/common/VInput.vue";
import VCheckbox from "@/components/common/VCheckbox.vue";

import EncryptIcon from "@/../static/images/objects/encrypt.svg";
import InfoIcon from "@/../static/images/common/smallGreyInfo.svg";

// @vue/component
@Component({
    components: {
        EncryptIcon,
        InfoIcon,
        VInfo,
        VButton,
        VInput,
        VCheckbox,
    },
})
export default class GeneratePassphrase extends Vue {
    @Prop({ default: () => () => null })
    public readonly onNextClick: () => unknown;
    @Prop({ default: () => () => null })
    public readonly onBackClick: () => unknown;
    @Prop({ default: () => () => null })
    public readonly setParentPassphrase: (passphrase: string) => void;
    @Prop({ default: false })
    public readonly isLoading: boolean;

    public isGenerate = true;
    public enterError = '';
    public passphrase = '';
    public isSavingConfirmed = false;
    public isCheckboxError = false;

    /**
     * Lifecycle hook after initial render.
     * Chooses correct state and generates mnemonic.
     */
    public mounted(): void {
        const idPassSalt: UserIDPassSalt | null = LocalData.getUserIDPassSalt();
        if (idPassSalt && idPassSalt.userId === this.$store.getters.user.id) {
            this.isGenerate = false;

            return;
        }

        this.passphrase = generateMnemonic();
        this.setParentPassphrase(this.passphrase);
    }

    public setSavingConfirmation(value: boolean): void {
        this.isSavingConfirmed = value;
    }

    /**
     * Holds on copy button click logic.
     * Copies passphrase to clipboard.
     */
    public onCopyClick(): void {
        this.$copyText(this.passphrase);
        this.$notify.success('Passphrase was copied successfully');
    }

    /**
     * Holds on download button click logic.
     * Downloads encryption passphrase as a txt file.
     */
    public onDownloadClick(): void {
        if (!this.passphrase) {
            this.enterError = 'Can\'t be empty!';

            return;
        }

        const fileName = 'StorjEncryptionPassphrase.txt';

        Download.file(this.passphrase, fileName);
    }

    /**
     * Sets passphrase from child component.
     */
    public setPassphrase(passphrase: string): void {
        if (this.enterError) this.enterError = '';

        this.passphrase = passphrase;
        this.setParentPassphrase(this.passphrase);
    }

    /**
     * Sets view state to enter passphrase.
     */
    public setToEnter(): void {
        this.passphrase = '';
        this.setParentPassphrase(this.passphrase);
        this.isGenerate = false;
    }

    /**
     * Sets view state to generate passphrase.
     */
    public setToGenerate(): void {
        if (this.enterError) this.enterError = '';

        this.passphrase = generateMnemonic();
        this.setParentPassphrase(this.passphrase);
        this.isGenerate = true;
    }

    /**
     * Holds on next button click logic.
     */
    public async onNextButtonClick(): Promise<void> {
        if (!this.passphrase) {
            this.enterError = 'Can\'t be empty!';

            return;
        }

        if (!this.isSavingConfirmed) {
            this.isCheckboxError = true;

            return;
        }

        await this.onNextClick();
    }

    /**
     * Returns objects flow status from store.
     */
    public get isNewObjectsFlow(): string {
        return this.$store.state.appStateModule.isNewObjectsFlow;
    }
}
</script>

<style scoped lang="scss">
    .encrypt-container {
        font-family: 'font_regular', sans-serif;
        padding: 40px 60px 60px;
        max-width: 500px;
        background: #fcfcfc;
        box-shadow: 0 0 32px rgb(0 0 0 / 4%);
        border-radius: 20px;
        margin: 30px auto 0;
        display: flex;
        flex-direction: column;
        align-items: center;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 36px;
            line-height: 56px;
            letter-spacing: 1px;
            color: #14142b;
            margin: 10px 0;
        }

        &__info {
            font-size: 16px;
            line-height: 28px;
            letter-spacing: 0.75px;
            color: #1b2533;
            margin-bottom: 20px;
            text-align: center;
            max-width: 420px;
        }

        &__functional {
            border: 1px solid #e6e9ef;
            border-radius: 10px;
            padding: 20px 0;

            &__header {
                width: calc(100% - 50px);
                display: flex;
                align-items: center;
                justify-content: space-between;
                padding: 0 25px;
                border-bottom: 1px solid #e6e9ef;

                &__gen {
                    font-family: 'font_medium', sans-serif;
                    font-size: 14px;
                    line-height: 17px;
                    color: #a9b5c1;
                    padding-bottom: 20px;
                    border-bottom: 4px solid #fff;
                    cursor: pointer;
                    white-space: nowrap;
                }

                &__right {
                    display: flex;
                    align-items: flex-start;
                    padding-bottom: 20px;
                    border-bottom: 4px solid #fff;
                    cursor: pointer;

                    &__enter {
                        font-family: 'font_medium', sans-serif;
                        font-size: 14px;
                        line-height: 17px;
                        color: #a9b5c1;
                        margin-right: 10px;
                        white-space: nowrap;
                    }

                    &__info-button {

                        &__image {
                            cursor: pointer;
                        }

                        &__message {
                            color: #586c86;
                            font-size: 16px;
                            line-height: 21px;
                        }
                    }
                }
            }

            &__generate {
                display: flex;
                align-items: center;
                padding: 16px 22px;
                background: #eff0f7;
                border-radius: 10px;
                margin: 25px 25px 0;

                &__value {
                    font-size: 14px;
                    line-height: 25px;
                    color: #384b65;
                }

                &__button {
                    margin-left: 32px;
                    min-width: 66px;
                }
            }

            &__enter {
                margin: 25px 25px 0;
            }

            &__download {
                font-family: 'font_bold', sans-serif;
                font-size: 16px;
                line-height: 19px;
                color: #0068dc;
                cursor: pointer;
                margin: 20px 25px;
                display: inline-block;
            }

            &__warning-title {
                font-family: 'font_bold', sans-serif;
                font-size: 16px;
                line-height: 19px;
                color: #1b2533;
                margin: 25px 25px 10px;
            }

            &__warning-msg {
                font-size: 14px;
                line-height: 20px;
                color: #1b2533;
                margin: 0 25px;
            }

            &__checkbox {
                margin: 0 25px;
            }
        }

        &__buttons {
            width: 100%;
            display: flex;
            align-items: center;
            margin-top: 30px;

            &__back {
                margin-right: 24px;
            }
        }
    }

    .active {
        color: #0149ff;
        border-color: #0149ff;
    }

    .active svg rect {
        fill: #0149ff;
    }

    ::v-deep .info__box__message {
        min-width: 440px;
    }
</style>
