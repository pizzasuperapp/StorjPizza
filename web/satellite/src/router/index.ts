// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router from 'vue-router';

import AccessGrants from '@/components/accessGrants/AccessGrants.vue';
import CreateAccessGrant from '@/components/accessGrants/CreateAccessGrant.vue';
import CLIStep from '@/components/accessGrants/steps/CLIStep.vue';
import CreatePassphraseStep from '@/components/accessGrants/steps/CreatePassphraseStep.vue';
import EnterPassphraseStep from '@/components/accessGrants/steps/EnterPassphraseStep.vue';
import GatewayStep from '@/components/accessGrants/steps/GatewayStep.vue';
import NameStep from '@/components/accessGrants/steps/NameStep.vue';
import PermissionsStep from '@/components/accessGrants/steps/PermissionsStep.vue';
import ResultStep from '@/components/accessGrants/steps/ResultStep.vue';
import AccountArea from '@/components/account/AccountArea.vue';
import AccountBilling from '@/components/account/billing/BillingArea.vue';

// new payment screen routes
import BillingOverview from '@/components/account/billing/billingTabs/Overview.vue';
import BillingPaymentMethods from '@/components/account/billing/billingTabs/PaymentMethods.vue';
import BillingHistory2 from '@/components/account/billing/billingTabs/BillingHistory.vue';
import BillingCoupons from '@/components/account/billing/billingTabs/Coupons.vue';


import DetailedHistory from '@/components/account/billing/depositAndBillingHistory/DetailedHistory.vue';
import AddCouponCode from '@/components/account/billing/coupons/AddCouponCode.vue';
import CreditsHistory from '@/components/account/billing/coupons/CouponArea.vue';
import SettingsArea from '@/components/account/SettingsArea.vue';
import Page404 from '@/components/errors/Page404.vue';
import BucketsView from '@/components/objects/BucketsView.vue';
import EncryptData from '@/components/objects/EncryptData.vue';
import ObjectsArea from '@/components/objects/ObjectsArea.vue';
import UploadFile from '@/components/objects/UploadFile.vue';
import OnboardingTourArea from '@/components/onboardingTour/OnboardingTourArea.vue';
import OnbCLIStep from '@/components/onboardingTour/steps/CLIStep.vue';
import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';
import CreateProject from '@/components/project/CreateProject.vue';
import EditProjectDetails from '@/components/project/EditProjectDetails.vue';
import ProjectDashboard from '@/components/project/ProjectDashboard.vue';
import NewProjectDashboard from "@/components/project/newProjectDashboard/NewProjectDashboard.vue";
import ProjectsList from '@/components/projectsList/ProjectsList.vue';
import ProjectMembersArea from '@/components/team/ProjectMembersArea.vue';
import CLIInstall from "@/components/onboardingTour/steps/cliFlow/CLIInstall.vue";
import APIKey from "@/components/onboardingTour/steps/cliFlow/APIKey.vue";
import CLISetup from "@/components/onboardingTour/steps/cliFlow/CLISetup.vue";
import CreateBucket from "@/components/onboardingTour/steps/cliFlow/CreateBucket.vue";
import UploadObject from "@/components/onboardingTour/steps/cliFlow/UploadObject.vue";
import ListObject from "@/components/onboardingTour/steps/cliFlow/ListObject.vue";
import DownloadObject from "@/components/onboardingTour/steps/cliFlow/DownloadObject.vue";
import ShareObject from "@/components/onboardingTour/steps/cliFlow/ShareObject.vue";
import RegistrationSuccess from "@/components/common/RegistrationSuccess.vue";
import SuccessScreen from "@/components/onboardingTour/steps/cliFlow/SuccessScreen.vue";
import AGName from "@/components/onboardingTour/steps/cliFlow/AGName.vue";
import AGPermissions from "@/components/onboardingTour/steps/cliFlow/AGPermissions.vue";
import BucketCreation from "@/components/objects/BucketCreation.vue";

import { NavigationLink } from '@/types/navigation';
import { MetaUtils } from "@/utils/meta";
import BucketDetails from "@/components/objects/BucketDetails.vue";

const ActivateAccount = () => import('@/views/ActivateAccount.vue');
const AuthorizeArea = () => import('@/views/AuthorizeArea.vue');
const DashboardArea = () => import('@/views/DashboardArea.vue');
const ForgotPassword = () => import('@/views/ForgotPassword.vue');
const LoginArea = () => import('@/views/LoginArea.vue');
const RegisterArea = () => import('@/views/RegisterArea.vue');
const ResetPassword = () => import('@/views/ResetPassword.vue');

Vue.use(Router);

/**
 * RouteConfig contains information about all routes and subroutes
 */
export abstract class RouteConfig {
    // root paths
    public static Root = new NavigationLink('/', 'Root');
    public static Login = new NavigationLink('/login', 'Login');
    public static Register = new NavigationLink('/signup', 'Register');
    public static RegisterSuccess = new NavigationLink('/signup-success', 'RegisterSuccess');
    public static Activate = new NavigationLink('/activate', 'Activate');
    public static ForgotPassword = new NavigationLink('/forgot-password', 'Forgot Password');
    public static ResetPassword = new NavigationLink('/password-recovery', 'Reset Password');
    public static Authorize = new NavigationLink('/oauth/v2/authorize', 'Authorize')
    public static Account = new NavigationLink('/account', 'Account');
    public static ProjectDashboard = new NavigationLink('/project-dashboard', 'Project Overview');
    public static NewProjectDashboard = new NavigationLink('/new-project-dashboard', 'Project Overview ');
    public static Users = new NavigationLink('/project-members', 'Users');
    public static OnboardingTour = new NavigationLink('/onboarding-tour', 'Onboarding Tour');
    public static CreateProject = new NavigationLink('/create-project', 'Create Project');
    public static EditProjectDetails = new NavigationLink('/edit-project-details', 'Edit Project Details');
    public static AccessGrants = new NavigationLink('/access-grants', 'Access');
    public static ProjectsList = new NavigationLink('/projects', 'Projects');
    public static Buckets = new NavigationLink('/buckets', 'Buckets');

    // account child paths
    public static Settings = new NavigationLink('settings', 'Settings');
    public static Billing = new NavigationLink('billing', 'Billing');
    public static AddCouponCode = new NavigationLink('add-coupon', 'Get Free Credits');
    public static BillingHistory = new NavigationLink('billing-history', 'Billing History');
    public static BillingOverview = new NavigationLink('overview', 'Overview');
    public static BillingPaymentMethods = new NavigationLink('payment-methods', 'Payment Methods');
    public static BillingHistory2 = new NavigationLink('billing-history2', 'Billing History 2');
    public static BillingCoupons = new NavigationLink('coupons', 'Coupons');
    public static DepositHistory = new NavigationLink('deposit-history', 'Deposit History');
    public static CreditsHistory = new NavigationLink('credits-history', 'Credits History');

    // access grant child paths
    public static CreateAccessGrant = new NavigationLink('create-grant', 'Create Access Grant');
    public static NameStep = new NavigationLink('name', 'Name Access Grant');
    public static PermissionsStep = new NavigationLink('permissions', 'Access Grant Permissions');
    public static CreatePassphraseStep = new NavigationLink('create-passphrase', 'Access Grant Create Passphrase');
    public static EnterPassphraseStep = new NavigationLink('enter-passphrase', 'Access Grant Enter Passphrase');
    public static ResultStep = new NavigationLink('result', 'Access Grant Result');
    public static GatewayStep = new NavigationLink('gateway', 'Access Grant Gateway');
    public static CLIStep = new NavigationLink('cli', 'Access Grant In CLI');

    // onboarding tour child paths
    public static OverviewStep = new NavigationLink('overview', 'Onboarding Overview');
    public static OnbCLIStep = new NavigationLink('cli', 'Onboarding CLI');
    public static AGName = new NavigationLink('ag-name', 'Onboarding AG Name');
    public static AGPermissions = new NavigationLink('ag-permissions', 'Onboarding AG Permissions');
    public static APIKey = new NavigationLink('api-key', 'Onboarding API Key');
    public static CLIInstall = new NavigationLink('cli-install', 'Onboarding CLI Install');
    public static CLISetup = new NavigationLink('cli-setup', 'Onboarding CLI Setup');
    public static CreateBucket = new NavigationLink('create-bucket', 'Onboarding Create Bucket');
    public static UploadObject = new NavigationLink('upload-object', 'Onboarding Upload Object');
    public static ListObject = new NavigationLink('list-object', 'Onboarding List Object');
    public static DownloadObject = new NavigationLink('download-object', 'Onboarding Download Object');
    public static ShareObject = new NavigationLink('share-object', 'Onboarding Share Object');
    public static SuccessScreen = new NavigationLink('success', 'Onboarding Success Screen');

    // objects child paths.
    public static EncryptData = new NavigationLink('encrypt-data', 'Objects Encrypt Data');
    public static BucketsManagement = new NavigationLink('management', 'Buckets Management');
    public static BucketsDetails = new NavigationLink('details', 'Bucket Details');
    public static UploadFile = new NavigationLink('upload/', 'Objects Upload');
    public static UploadFileChildren = new NavigationLink('*', 'Objects Upload Children');
    public static BucketCreation = new NavigationLink('creation', 'Bucket Creation')
}

const isNewProjectDashboard = MetaUtils.getMetaContent('new-project-dashboard') === 'true';
if (isNewProjectDashboard) {
    RouteConfig.ProjectDashboard = RouteConfig.NewProjectDashboard
}

export const notProjectRelatedRoutes = [
    RouteConfig.Login.name,
    RouteConfig.Register.name,
    RouteConfig.RegisterSuccess.name,
    RouteConfig.Activate.name,
    RouteConfig.ForgotPassword.name,
    RouteConfig.ResetPassword.name,
    RouteConfig.Authorize.name,
    RouteConfig.Billing.name,
    RouteConfig.BillingHistory.name,
    RouteConfig.BillingOverview.name,
    RouteConfig.BillingPaymentMethods.name,
    RouteConfig.BillingHistory2.name,
    RouteConfig.BillingCoupons.name,
    RouteConfig.DepositHistory.name,
    RouteConfig.CreditsHistory.name,
    RouteConfig.Settings.name,
];

export const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Login.path,
            name: RouteConfig.Login.name,
            component: LoginArea,
        },
        {
            path: RouteConfig.Register.path,
            name: RouteConfig.Register.name,
            component: RegisterArea,
        },
        {
            path: RouteConfig.RegisterSuccess.path,
            name: RouteConfig.RegisterSuccess.name,
            component: RegistrationSuccess,
        },
        {
            path: RouteConfig.Activate.path,
            name: RouteConfig.Activate.name,
            component: ActivateAccount,
        },
        {
            path: RouteConfig.ForgotPassword.path,
            name: RouteConfig.ForgotPassword.name,
            component: ForgotPassword,
        },
        {
            path: RouteConfig.ResetPassword.path,
            name: RouteConfig.ResetPassword.name,
            component: ResetPassword,
        },
        {
            path: RouteConfig.Authorize.path,
            name: RouteConfig.Authorize.name,
            component: AuthorizeArea,
        },
        {
            path: RouteConfig.Root.path,
            meta: {
                requiresAuth: true,
            },
            component: DashboardArea,
            children: [
                {
                    path: RouteConfig.Root.path,
                    name: 'default',
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Account.path,
                    name: RouteConfig.Account.name,
                    component: AccountArea,
                    children: [
                        {
                            path: RouteConfig.Settings.path,
                            name: RouteConfig.Settings.name,
                            component: SettingsArea,
                        },
                        {
                            path: RouteConfig.Billing.path,
                            name: RouteConfig.Billing.name,
                            component: AccountBilling,
                            children: [
                                {
                                    path: RouteConfig.AddCouponCode.path,
                                    name: RouteConfig.AddCouponCode.name,
                                    component: AddCouponCode,
                                },
                                {
                                    path: RouteConfig.BillingOverview.path,
                                    name: RouteConfig.BillingOverview.name,
                                    component: BillingOverview,
                                },
                                {
                                    path: RouteConfig.BillingPaymentMethods.path,
                                    name: RouteConfig.BillingPaymentMethods.name,
                                    component: BillingPaymentMethods,
                                },
                                {
                                    path: RouteConfig.BillingHistory2.path,
                                    name: RouteConfig.BillingHistory2.name,
                                    component: BillingHistory2,
                                },
                                {
                                    path: RouteConfig.BillingCoupons.path,
                                    name: RouteConfig.BillingCoupons.name,
                                    component: BillingCoupons,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BillingHistory.path,
                            name: RouteConfig.BillingHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.DepositHistory.path,
                            name: RouteConfig.DepositHistory.name,
                            component: DetailedHistory,
                        },
                        {
                            path: RouteConfig.CreditsHistory.path,
                            name: RouteConfig.CreditsHistory.name,
                            component: CreditsHistory,
                        },
                    ],
                },
                {
                    path: RouteConfig.NewProjectDashboard.path,
                    name: RouteConfig.NewProjectDashboard.name,
                    component: NewProjectDashboard,
                },
                {
                    path: RouteConfig.ProjectDashboard.path,
                    name: RouteConfig.ProjectDashboard.name,
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Users.path,
                    name: RouteConfig.Users.name,
                    component: ProjectMembersArea,
                },
                {
                    path: RouteConfig.OnboardingTour.path,
                    name: RouteConfig.OnboardingTour.name,
                    component: OnboardingTourArea,
                    children: [
                        {
                            path: RouteConfig.OverviewStep.path,
                            name: RouteConfig.OverviewStep.name,
                            component: OverviewStep,
                        },
                        {
                            path: RouteConfig.OnbCLIStep.path,
                            name: RouteConfig.OnbCLIStep.name,
                            component: OnbCLIStep,
                            children: [
                                {
                                    path: RouteConfig.AGName.path,
                                    name: RouteConfig.AGName.name,
                                    component: AGName,
                                },
                                {
                                    path: RouteConfig.AGPermissions.path,
                                    name: RouteConfig.AGPermissions.name,
                                    component: AGPermissions,
                                },
                                {
                                    path: RouteConfig.APIKey.path,
                                    name: RouteConfig.APIKey.name,
                                    component: APIKey,
                                },
                                {
                                    path: RouteConfig.CLIInstall.path,
                                    name: RouteConfig.CLIInstall.name,
                                    component: CLIInstall,
                                },
                                {
                                    path: RouteConfig.CLISetup.path,
                                    name: RouteConfig.CLISetup.name,
                                    component: CLISetup,
                                },
                                {
                                    path: RouteConfig.CreateBucket.path,
                                    name: RouteConfig.CreateBucket.name,
                                    component: CreateBucket,
                                },
                                {
                                    path: RouteConfig.UploadObject.path,
                                    name: RouteConfig.UploadObject.name,
                                    component: UploadObject,
                                },
                                {
                                    path: RouteConfig.ListObject.path,
                                    name: RouteConfig.ListObject.name,
                                    component: ListObject,
                                },
                                {
                                    path: RouteConfig.DownloadObject.path,
                                    name: RouteConfig.DownloadObject.name,
                                    component: DownloadObject,
                                },
                                {
                                    path: RouteConfig.ShareObject.path,
                                    name: RouteConfig.ShareObject.name,
                                    component: ShareObject,
                                },
                                {
                                    path: RouteConfig.SuccessScreen.path,
                                    name: RouteConfig.SuccessScreen.name,
                                    component: SuccessScreen,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: RouteConfig.CreateProject.path,
                    name: RouteConfig.CreateProject.name,
                    component: CreateProject,
                },
                {
                    path: RouteConfig.EditProjectDetails.path,
                    name: RouteConfig.EditProjectDetails.name,
                    component: EditProjectDetails,
                },
                {
                    path: RouteConfig.AccessGrants.path,
                    name: RouteConfig.AccessGrants.name,
                    component: AccessGrants,
                    children: [
                        {
                            path: RouteConfig.CreateAccessGrant.path,
                            name: RouteConfig.CreateAccessGrant.name,
                            component: CreateAccessGrant,
                            children: [
                                {
                                    path: RouteConfig.NameStep.path,
                                    name: RouteConfig.NameStep.name,
                                    component: NameStep,
                                },
                                {
                                    path: RouteConfig.PermissionsStep.path,
                                    name: RouteConfig.PermissionsStep.name,
                                    component: PermissionsStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.CreatePassphraseStep.path,
                                    name: RouteConfig.CreatePassphraseStep.name,
                                    component: CreatePassphraseStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.EnterPassphraseStep.path,
                                    name: RouteConfig.EnterPassphraseStep.name,
                                    component: EnterPassphraseStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.ResultStep.path,
                                    name: RouteConfig.ResultStep.name,
                                    component: ResultStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.GatewayStep.path,
                                    name: RouteConfig.GatewayStep.name,
                                    component: GatewayStep,
                                    props: true,
                                },
                                {
                                    path: RouteConfig.CLIStep.path,
                                    name: RouteConfig.CLIStep.name,
                                    component: CLIStep,
                                    props: true,
                                },
                            ],
                        },
                    ],
                },
                {
                    path: RouteConfig.ProjectsList.path,
                    name: RouteConfig.ProjectsList.name,
                    component: ProjectsList,
                },
                {
                    path: RouteConfig.Buckets.path,
                    name: RouteConfig.Buckets.name,
                    component: ObjectsArea,
                    children: [
                        {
                            path: RouteConfig.EncryptData.path,
                            name: RouteConfig.EncryptData.name,
                            component: EncryptData,
                        },
                        {
                            path: RouteConfig.BucketsManagement.path,
                            name: RouteConfig.BucketsManagement.name,
                            component: BucketsView,
                            props: true,
                        },
                        {
                            path: RouteConfig.BucketsDetails.path,
                            name: RouteConfig.BucketsDetails.name,
                            component: BucketDetails,
                            props: true,
                        },
                        {
                            path: RouteConfig.UploadFile.path,
                            name: RouteConfig.UploadFile.name,
                            component: UploadFile,
                            children: [
                                {
                                    path: RouteConfig.UploadFileChildren.path,
                                    name: RouteConfig.UploadFileChildren.name,
                                    component: UploadFile,
                                },
                            ],
                        },
                        {
                            path: RouteConfig.BucketCreation.path,
                            name: RouteConfig.BucketCreation.name,
                            component: BucketCreation,
                        },
                    ],
                },
            ],
        },
        {
            path: '*',
            name: '404',
            component: Page404,
        },
    ],
});
