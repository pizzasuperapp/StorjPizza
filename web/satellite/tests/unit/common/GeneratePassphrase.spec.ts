// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import GeneratePassphrase from '@/components/common/GeneratePassphrase.vue';

import {createLocalVue, shallowMount} from '@vue/test-utils';

import Vuex from "vuex";
import {router} from "@/router";
import {appStateModule} from "@/store/modules/appState";

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({ modules: {
    appStateModule,
}});

describe('GeneratePassphrase.vue', () => {
    it('renders correctly with default props', () => {
        const wrapper = shallowMount<GeneratePassphrase>(GeneratePassphrase, {
            localVue,
            router,
            store,
        });

        expect(wrapper.vm.passphrase).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });
});
