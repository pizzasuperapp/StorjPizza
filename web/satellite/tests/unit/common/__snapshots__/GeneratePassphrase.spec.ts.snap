// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneratePassphrase.vue renders correctly with default props 1`] = `
<div class="encrypt-container">
  <encrypticon-stub></encrypticon-stub>
  <h1 aria-roledescription="enc-title" class="encrypt-container__title">Encryption passphrase</h1>
  <p class="encrypt-container__info">
    The encryption passphrase is used to encrypt and access the data that you upload to Storj.
  </p>
  <div class="encrypt-container__functional">
    <div class="encrypt-container__functional__header">
      <p class="encrypt-container__functional__header__gen active">
        Generate a new passphrase
      </p>
      <div class="encrypt-container__functional__header__right">
        <p aria-roledescription="enter-passphrase-label" class="encrypt-container__functional__header__right__enter">
          Enter your own passphrase
        </p>
        <vinfo-stub title="" buttonlabel="" onbuttonclick="[Function]" class="encrypt-container__functional__header__right__info-button"></vinfo-stub>
      </div>
    </div>
    <div class="encrypt-container__functional__generate">
      <p class="encrypt-container__functional__generate__value"></p>
      <vbutton-stub label="Copy" width="66px" height="30px" fontsize="16px" borderradius="6px" isbluewhite="true" isuppercase="true" onpress="[Function]" class="encrypt-container__functional__generate__button"></vbutton-stub>
    </div>
    <h2 aria-roledescription="warning-title" class="encrypt-container__functional__warning-title">
      Save your encryption passphrase
    </h2>
    <p class="encrypt-container__functional__warning-msg">
      Please note that Storj does not know or store your encryption passphrase. If you lose it, you will not
      be able to recover your files.
    </p>
    <p class="encrypt-container__functional__download">Download as a text file</p>
    <vcheckbox-stub label="I understand, and I have saved the passphrase." class="encrypt-container__functional__checkbox"></vcheckbox-stub>
  </div>
  <div class="encrypt-container__buttons">
    <!---->
    <vbutton-stub label="Next >" width="inherit" height="64px" fontsize="16px" borderradius="62px" isdisabled="true" onpress="[Function]"></vbutton-stub>
  </div>
</div>
`;
